packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -102,6 +106,10 @@ export default function NavigationPlaceholder( {
 		blocks,
 		navigationMenuTitle = null
 	) => {
+		if ( ! canUserCreateNavigation ) {
```

Comment:
It makes sense to do it this way in this PR, but I'd like to get to a place where the flow of data is easy to understand and we can just be sure this function is not called.

Reply:
Agreed.